manifests/state-container-networking-plugins/0010-container-networking-plugins-ds.yml

```diff
@@ -55,13 +55,30 @@ spec:
           imagePullPolicy: IfNotPresent
           securityContext:
             privileged: true
+          {{- $containerResourcesFound := false }}
```

Comment:
Maybe we can try to create a template function which will contain this logic
something like 
`addResources(.CrSpec.ContainerResources,  "cni-plugins",  "100m", "50Mi", "100m", 50Mi")`
or 
`addResources(.CrSpec.ContainerResources,  "cni-plugins",  objectWithDefaults)`

Reply:
i think, if we use for the template a map keyed by container name then we can check for key existance then with if/else use req/limits from map entry or default as provided by template.

for those templates that dont need default then we drop the "else" part

thoughts ?


presonally;
1. i want to avoid inventing sort of a DSL with special functions
2. i dont want to move template information to code (e.g defaults will be part of state)